Formula/c/chatgpt-cli.rb

```diff
@@ -0,0 +1,33 @@
+class ChatgptCli < Formula
+  desc "CLI for interacting with the OpenAI ChatGPT API"
+  homepage "https://github.com/kardolus/chatgpt-cli"
+  url "https://github.com/kardolus/chatgpt-cli/archive/refs/tags/v1.5.5.tar.gz"
+  sha256 "158620e871411053310c6882a3e164e013389b0b6d99dd6fb42d8d82c988e51d"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    git_commit = "422f3b9dc0b290ed7b7bc86d1d6c83ff79266de0"
+
+    ldflags = %W[
+      -s -w
+      -X main.GitCommit=#{git_commit}
```

Comment:
```suggestion
      -X main.GitCommit=#{tap.user}
```

Reply:
@chenrui333 thanks. I tested the `#{tap.user}` variable and here is the output of my version command: 
```
 2024-04-21 09:04:51 ⌚  Guillermos-MacBook-Pro in ~/workspace/homebrew-core
± |master U:1 ✗| → chatgpt --version
ChatGPT CLI version v1.5.5 (commit local)
```
It was: 
```
 2024-04-21 08:38:59 ⌚  Guillermos-MacBook-Pro in ~/workspace/homebrew-core
± |master U:1 ✗| → chatgpt --version
ChatGPT CLI version 1.5.5 (commit 422f3b9dc0b290ed7b7bc86d1d6c83ff79266de0)
```

This will be a breaking change. I would have to get rid of the "commit" word from the version command and then bump the version to the `1.6` line since folks may be reliant on the git hash being there. It will make potential debugging more painful on my end as well since I cannot quickly checkout the hash in order to reproduce the issue. 

One work around could be to hardcode the git hash in a text file, and then read it from the brew script. We could also leverage `Utils.git_head`. Would not be perfect though.

Reply:
tap.user works for me, and that is the one used for many other formulae too.

Reply:
> since folks may be reliant on the git hash being there

I dont think so.

Reply:
> It will make potential debugging more painful on my end as well since I cannot quickly checkout the hash in order to reproduce the issue.

Like I said earlier, the github release tag maps to the github hash (if you check out tag, you get the corresponding github commit)

Reply:
@chenrui333 that is not true if you compile straight from the main branch or from your own fork. In that case the hash and the tag may not match. I am not going to assume what folks may or may not be reliant on. A breaking change means a version bump. I know that folks at Stripe are using the CLI for example. What if they pull my hash from that command. Who knows? Can't really assume anything that happens downstream imo.

Reply:
@chenrui333 there will be users who will not use homebrew. These folks are encouraged to use the installation scripts provided with the CLI. Those scripts currently add the git hash and the latest tag. Those may not be the same as the hash/tag from the release.

I am still trying to wrap my head around a solution. One way would be the hardcoded hash in a text file. Or maybe some string like "homebrew compiled" or something. Just so I can match the git hash with the issue a user may encounter if that makes sense. 

Thanks for helping out btw. Super appreciate it. Learned a bunch :)

Reply:
@chenrui333 Perhaps something like `ChatGPT CLI version v1.5.5 (homebrew)` for people who use `brew` and `ChatGPT CLI version 1.5.5 (commit 422f3b9dc0b290ed7b7bc86d1d6c83ff79266de0)` for people who do not use `brew`?

I can make this change, but it will require a version bump (`1.5.6`). Let me know what you think or if you have other ideas.